### PR TITLE
docker: useful ping error message

### PR DESCRIPTION
**Pull request by runcom**
In reply to https://github.com/projectatomic/docker/issues/214#issuecomment-263937593

@mtrmac PTAL

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

**Comment by mtrmac** [comment]
Well really `json: cannot unmarshal bool into Go value of type docker.APIErrors` should be handled internally by doing $something_else with the response body, but this is a clear improvement.

:+1: pending tests.

**Comment by runcom** [comment]
@mtrmac should we check for that string and warn that's a V1 registry clearer?

**Comment by runcom** [comment]
@mtrmac PTAL, I modified this patch so that we now have:
```sh
 ./skopeo inspect docker://quay.io/pypa/manylinux1_x86_64
FATA[0003] can't talk to a V1 docker registry
```

**Comment by wallrj** [comment]
@runcom That looks like a great improvement, It'd be even clearer if it said:
> Unable to connect to V1 Docker registry because `signature-verification` is enabled."

**Comment by runcom** [comment]
> @runcom That looks like a great improvement, It'd be even clearer if it said:
> 
> Unable to connect to V1 Docker registry because signature-verification is enabled."

@wallrj yup, that will be fixed in `projectatomic/docker` once this PR gets in :+1:

**Comment by runcom** [comment]
> Thanks, this is great. Just one nit.

@mtrmac Fixed, will wait for tests and then merge (then hopefully update docker to check that error to return a proper error message in the docker cli), thanks

**Comment by runcom** [comment]
lgtm

**Comment by runcom** [comment]
@wallrj docker will look like:
```sh
$ docker pull quay.io/pypa/manylinux1_x86_64
Using default tag: latest
Trying to pull repository quay.io/pypa/manylinux1_x86_64 ...
unable to pull from V1 Docker registries with image signatures verification enabled, please set --signature-enabled=false in the docker daemon
```

**Comment by wallrj** [comment]
Thanks @runcom 
Ping me when you submit the PR with the improved error message and I'll review it if you like.

**Comment by runcom** [comment]
@wallrj https://github.com/projectatomic/docker/commit/bc1668737f88c1f3a65b5aec945bcf00556d326d

**Comment by rhatdan** [comment]
Without looking at the code, I have a couple of comments on the message.

unable to pull from V1 Docker registries with image signatures verification enabled, please set --signature-enabled=false in the docker daemon

Can we say V1 Container Registries?
Also does the --signature-enabled message from docker or from containers/image?  If from containers image, that is a mistake since we might be doing this outside of docker daemon.
Lastly this gets the user to disable a security feature.

I would prefer something to state that only pull v2 images if you want signature verification. If V1 images required then --signature-enabled=false.

**Comment by runcom** [comment]
> Also does the --signature-enabled message from docker or from containers/image? If from containers image, that is a mistake since we might be doing this outside of docker daemon.
> Lastly this gets the user to disable a security feature.

that message will come from docker

> I would prefer something to state that only pull v2 images if you want signature verification. If V1 images required then --signature-enabled=false.

@rhatdan can you rephrase it to be an error message? I might be getting it wrong again cause I'm not a native speaker.

> Can we say V1 Container Registries?

I don't see any issues.

**Comment by mtrmac** [comment]
Yeah, the wording could be a little more scary :)  Something like

`unable to pull from V1 Docker registries with image signature verification enabled. If you need to accept this risk and disable signature verification (for ALL images), run the docker daemon with --signature-enabled=false`

? Though that is too long I guess.

**Comment by runcom** [comment]
> ? Though that is too long I guess.

It's fine to me, at least we clarify the situation further and that's good :)

**Comment by rhatdan** [comment]
Runcom as @mtrmac My main point was the error message says something like "Can't do that, so disable the security"  As opposed to Can't do that, use a different image or if you absolutely need this image do this at your own risk.

**Comment by runcom** [comment]
@rhatdan fair enough, I'll go with @mtrmac's suggestion if there are no objection.

**Comment by rhatdan** [comment]
SGTM
